cryostatio/cryostat-operator
# feat(tls): enable TLS on authproxies

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Based on #809
Depends on #809
Related to #710

## Description of the change:
*This change adds allows the users to provide...*

## Motivation for the change:
*This change is helpful because users may want to...*

## How to manually test:
1. Check out PR, build with `OPENSHIFT_OAUTH_PROXY_IMG=quay.io/andrewazores/openshift-oauth-proxy:test-14`. Deploy, then create a Cryostat CR. **Don't** set `DISABLE_SERVICE_TLS`, and **do** leave `enableCertManager:true`!
2. Verify that one Route is created pointing at port `4180`, which is the authproxy HTTP(S) port. It should redirect HTTP requests to HTTPS, and then it should reencrypt HTTPS traffic from the Router edge to the authproxy service.
3. Open `http://cryostat-cryostat.mycluster.domain` in your browser. Note that the URL protocol is `http`. The Route should immediately redirect you to HTTPS. You should then get to the OpenShift SSO login page. From this point on everything should behave as normal from prior PRs on this branch.
4. Check the auth-proxy container spec or logs and ensure that its `--http-address` is set to a blank value and `--https-address` is set to use port `4180`. If deploying oauth2_proxy, check its ConfigMap for this rather than its args.

To test oauth2_proxy:
1. deploy on non-OpenShift. Or, to force oauth2_proxy while on OpenShift, apply the following patch:
```diff
diff --git a/internal/controllers/common/resource_definitions/resource_definitions.go b/internal/controllers/common/resource_definitions/resource_definitions.go
index 018320e..8f0ffe0 100644
--- a/internal/controllers/common/resource_definitions/resource_definitions.go
+++ b/internal/controllers/common/resource_definitions/resource_definitions.go
@@ -352,26 +352,26 @@ func NewPodForCR(cr *model.CryostatInstance, specs *ServiceSpecs, imageTags *Ima
        }
        volumes = append(volumes, certVolume)
 
-       if !openshift {
-               // if not deploying openshift-oauth-proxy then we must be deploying oauth2_proxy instead
-               volumes = append(volumes, corev1.Volume{
-                       Name: cr.Name + "-oauth2-proxy-cfg",
-                       VolumeSource: corev1.VolumeSource{
-                               ConfigMap: &corev1.ConfigMapVolumeSource{
-                                       LocalObjectReference: corev1.LocalObjectReference{
-                                               Name: cr.Name + "-oauth2-proxy-cfg",
-                                       },
-                                       Items: []corev1.KeyToPath{
-                                               {
-                                                       Key:  OAuth2ConfigFileName,
-                                                       Path: OAuth2ConfigFileName,
-                                                       Mode: &readOnlyMode,
-                                               },
+       // if !openshift {
+       // if not deploying openshift-oauth-proxy then we must be deploying oauth2_proxy instead
+       volumes = append(volumes, corev1.Volume{
+               Name: cr.Name + "-oauth2-proxy-cfg",
+               VolumeSource: corev1.VolumeSource{
+                       ConfigMap: &corev1.ConfigMapVolumeSource{
+                               LocalObjectReference: corev1.LocalObjectReference{
+                                       Name: cr.Name + "-oauth2-proxy-cfg",
+                               },
+                               Items: []corev1.KeyToPath{
+                                       {
+                                               Key:  OAuth2ConfigFileName,
+                                               Path: OAuth2ConfigFileName,
+                                               Mode: &readOnlyMode,
                                        },
                                },
                        },
-               })
-       }
+               },
+       })
+       // }
 
        if isBasicAuthEnabled(cr) {
                volumes = append(volumes,
@@ -636,9 +636,9 @@ func NewAuthProxyContainerResource(cr *model.CryostatInstance) *corev1.ResourceR
 
 func NewAuthProxyContainer(cr *model.CryostatInstance, specs *ServiceSpecs, oauth2ProxyImageTag string, openshiftAuthProxyImageTag string,
        tls *TLSConfig, openshift bool) (*corev1.Container, error) {
-       if openshift {
-               return NewOpenShiftAuthProxyContainer(cr, specs, openshiftAuthProxyImageTag, tls)
-       }
+       // if openshift {
+       //      return NewOpenShiftAuthProxyContainer(cr, specs, openshiftAuthProxyImageTag, tls)
+       // }
        return NewOAuth2ProxyContainer(cr, specs, oauth2ProxyImageTag, tls)
 }
 
diff --git a/internal/controllers/configmaps.go b/internal/controllers/configmaps.go
index 660fb38..194debd 100644
--- a/internal/controllers/configmaps.go
+++ b/internal/controllers/configmaps.go
@@ -137,11 +137,11 @@ func (r *Reconciler) reconcileOAuth2ProxyConfig(ctx context.Context, cr *model.C
                Data:      data,
        }
 
-       if r.IsOpenShift {
-               return r.deleteConfigMap(ctx, cm)
-       } else {
-               return r.createOrUpdateConfigMap(ctx, cm, cr.Object)
-       }
+       // if r.IsOpenShift {
+       //      return r.deleteConfigMap(ctx, cm)
+       // } else {
+       return r.createOrUpdateConfigMap(ctx, cm, cr.Object)
+       // }
 }
```
5. rebuild and redeploy, repeat tests above
